src/System.Management.Automation/engine/Utils.cs

```diff
@@ -594,6 +594,47 @@ internal static string GetCurrentMajorVersion()
             return PSVersionInfo.PSVersion.Major.ToString(CultureInfo.InvariantCulture);
         }
 
+        /// <summary>
+        /// Checks whether the user's OS type is in a list of required types.
+        /// </summary>
+        /// <param name="requiredOSTypes">A list of required types.</param>
+        /// <returns>True if the user's OS type is in the list of required types, false otherwise.</returns>
+        internal static bool IsOSTypeValid(IEnumerable<string> requiredOSTypes)
+        {
+            string currentOSName = GetOSTypeString();
+            foreach (string requiredOSType in requiredOSTypes) 
+            {
+                if (requiredOSType.Equals(currentOSName, StringComparison.OrdinalIgnoreCase))
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        /// <summary>
+        /// Gets a string representation of the user's current OS version or null if unknown version.
+        /// </summary>
+        /// <returns>The string representation of the user's OS.</returns>
+        internal static string GetOSTypeString()
+        {
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX)) 
+            {
+                return "MacOS"; 
+            }
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
```

Comment:
The `else`s in this method aren't needed; once you enter an `if` you'll always return